File: frontend/src/app/features/admin/models/admin-support.models.ts
export interface ProcessingFailure {
  documentId: string;
  title: string;
  processingStatus: 'Failed';
  failureReason: string | null;
  failedAt: string;
}

export interface AuditLogEntry {
  auditLogEntryId: string;
  eventType: string;
  message: string;
  severity: string;
  userId: string | null;
  entityType: string | null;
  entityId: string | null;
  correlationId: string | null;
  createdAt: string;
}

export interface AuditLogFilters {
  from?: string;
  to?: string;
  eventType?: string;
}

File: src/KnowledgeOps.Application/Admin/AdminSupportModels.cs
namespace KnowledgeOps.Application.Admin;

public sealed record ProcessingFailureResult(
    Guid DocumentId,
    string Title,
    string ProcessingStatus,
    string? FailureReason,
    DateTimeOffset FailedAt);

public sealed record AuditLogQuery(
    DateTimeOffset? From,
    DateTimeOffset? To,
    string? EventType,
    int? Limit = null);

public sealed record AuditLogResult(
    Guid AuditLogEntryId,
    string EventType,
    string Message,
    string Severity,
    Guid? UserId,
    string? EntityType,
    Guid? EntityId,
    string? CorrelationId,
    DateTimeOffset CreatedAt);
